task/backend/analytical_storage.go

```diff
@@ -152,18 +152,44 @@ func (as *AnalyticalStorage) FindRuns(ctx context.Context, filter influxdb.RunFi
 		filterPart = fmt.Sprintf(`|> filter(fn: (r) => r.runID > %q)`, filter.After.String())
 	}
 
+	// creates flux script to filter based on time, if given
+	constructedTimeFilter := ""
+	parsedAfterTime := time.Time{}
+	parsedBeforeTime := time.Now()
+	if len(filter.AfterTime) > 0 {
+		parsedAfterTime, err = time.Parse(time.RFC3339, filter.AfterTime)
+		if err != nil {
+			return nil, 0, fmt.Errorf("failed parsing after time: %s", err.Error())
+		}
+	}
+	if len(filter.BeforeTime) > 0 {
+		parsedBeforeTime, err = time.Parse(time.RFC3339, filter.BeforeTime)
+		if err != nil {
+			return nil, 0, fmt.Errorf("failed parsing before time: %s", err.Error())
+		}
+
+	}
+
+	if parsedAfterTime.IsZero() || len(filter.BeforeTime) > 0 {
+		constructedTimeFilter = fmt.Sprintf(
+			`|> filter(fn: (r) =>time(v: r["scheduledFor"]) > %s and time(v: r["scheduledFor"]) < %s)`,
+			parsedAfterTime.Format(time.RFC3339),
+			parsedBeforeTime.Format(time.RFC3339))
+	}
```

Comment:
The check as written is essentially `if there is no AfterTime set or BeforeTime is set`. With your default zero value for `parsedAfterTime`, this check would always pass (negative time would return an error above. With your defaults, the constructed time filter (and resulting query) would be valid and could return valid data. However, if the user accidentally flips the times (something like `after = now()` and `before = now()-5m`), the query would never return any data, potentially leaving a confused user.

In order to provide meaningful feedback pre-emptively, you could modify the check and do something like:
```suggestion
	if !parsedBeforeTime.After(parsedAfterTime) {
		return nil, 0, errors.New("after time must be prior to before time") // or some other useful error
	}
	
	constructedTimeFilter = fmt.Sprintf(
		`|> filter(fn: (r) =>time(v: r["scheduledFor"]) > %s and time(v: r["scheduledFor"]) < %s)`,
		parsedAfterTime.Format(time.RFC3339),
		parsedBeforeTime.Format(time.RFC3339))
```

Or remove the check entirely

Reply:
I did a little combo of both. I realized my logic was wrong, but also saw how a user could give faulty data and mess up the constructed string